### ISSUE TITLE
Does d3fc support mobile devices?

### ISSUE THREAD

**Issue by zeleniy** [ScottLogic/d3fc#823]
If you check [highstock demos](http://www.highcharts.com/stock/demo) you will see that all charts are drawn perfect on the mobile phones. But [this d3fc example] (http://bl.ocks.org/ColinEberhardt/9667dae337dfd492f63c) - not. So does d3fc support mobile devices? And if not how i can mend this?

**Comment by chrisprice** [comment]
It should work, we're aware of some performance issues but rendering-wise
it should be fine. Certainly, it's rendering fine for me on android with
chrome 47.

**Comment by rymdluo** [comment]
I have tried my version of the barrel example on mobile and it does work, even pinch zooming to view a limited range.

CSS here is key.

**Comment by jleft** [comment]
For some inspiration... The [Wealth & Health of Nations example](http://d3fc.io/examples/wealth-and-health-of-nations/) is fairly mobile friendly [[source](https://github.com/ScottLogic/d3fc/tree/master/site/src/examples/wealth-and-health-of-nations)].

#816 - updating Wealth & Health of Nations example for better experience on mobile/tablet - may be useful too.

**Comment by zeleniy** [comment]
Bellow you can see [low barrel](http://d3fc.io/examples/low-barrel/) example in Firefox mobile mode. In general it looks fine, but i would like to know how to (automatically) decrease amount of tick's labels on volume charts. Because the are overlaps. Actually amount of labels should depends on .plot-area width and labels size. When you change size of the brush at navigation chart length of labels will change from hours (11:00, 12:00) to months - (November, December) which is much longer...

![screenshot from 2016-01-15 21-10-52](https://cloud.githubusercontent.com/assets/2780617/12361006/87fdc522-bbcc-11e5-9e36-60ea49148701.png)

**Comment by ColinEberhardt** [comment]
Thanks for clarifying, this relates to #573 - we do plan to add responsive functionality to the library at some point.

**Comment by chrisprice** [comment]
I've recently implemented responsive ticks for another project. As @zeleniy points out, the ideal number of ticks is a function of the amount of space available, the width of the ticks and the required spacing. However, as the tick width is derived from the tick format (which by default is dynamic) and any applied styling, I'm not sure d3fc will ever be able to provide a perfect solution.

I've found a decent solution is to set the following on each render -

```
  cartesianChart.xAxisTicks(container.layout('width') / constant);
```

Where `constant` is a context specific magic number which takes into account the behaviour of the [axis ticks method](https://github.com/mbostock/d3/wiki/SVG-Axes#ticks). N.B. the above also doesn't take into account the chart's left/right margins which may or may not be significant.
